zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/CommandDistributionDistributingV2Applier.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.engine.state.appliers;
+
+import io.camunda.zeebe.engine.state.TypedEventApplier;
+import io.camunda.zeebe.engine.state.mutable.MutableDistributionState;
+import io.camunda.zeebe.protocol.impl.record.value.distribution.CommandDistributionRecord;
+import io.camunda.zeebe.protocol.record.intent.CommandDistributionIntent;
+import org.agrona.Strings;
+
+public final class CommandDistributionDistributingV2Applier
+    implements TypedEventApplier<CommandDistributionIntent, CommandDistributionRecord> {
+
+  private final MutableDistributionState distributionState;
+
+  public CommandDistributionDistributingV2Applier(
+      final MutableDistributionState distributionState) {
+    this.distributionState = distributionState;
+  }
+
+  @Override
+  public void applyState(final long key, final CommandDistributionRecord value) {
+    if (Strings.isEmpty(value.getQueueId())) {
+      // added to pending state via {@link CommandDistributionEnqueuedApplier}
+      distributionState.addPendingDistribution(key, value.getPartitionId());
+    }
+    distributionState.addRetriableDistribution(key, value.getPartitionId());
```

Comment:
:question: is this change related to metrics? Could you explain why it's needed?

Reply:
Yes, otherwise it would be counted twice for enqueued distributions (see comment & commit message)